### PR TITLE
Indiana unified tax credit for the elderly

### LINKED ISSUE #2434: Indiana Unified Tax Credit for the Elderly
[Reference](https://forms.in.gov/Download.aspx?id=15394)

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #2434

### Summary
🧓💵🇺🇸

This pull request adds four parameter files for the unified tax credit for the elderly in Indiana, a state-level income tax credit for older filers. The parameter files define the eligibility criteria and the credit amounts based on income and filing status.

> _`unified_credit`_
> _Adding parameters for_
> _Indiana's aged_

### Walkthrough
*  Add parameter files for the unified tax credit for the elderly in Indiana ([link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-f57a0f95753fda19fbb6ea268cc5ed74763f7735ba320543931bc81c2228328cR1-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-ae0808f54eaad8ed98d212a5d3ebca73bc80756fefb74de115c8a9a498dfff97R1-R28), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-07e0286e73459a123765c8fb84126c56a35a86f4b8c04578aeb5d8804df9efd0L1-R27), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-bb8128753953cffe82c1a012ad1f7ed841fc234811b52f6dee5ece380c850242R1-R24))
  - Specify the minimum age of 65 required to claim the credit in `age_eligibility.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-f57a0f95753fda19fbb6ea268cc5ed74763f7735ba320543931bc81c2228328cR1-R11))
  - Define the credit amount as a function of adjusted gross income and filing status in `amount_married_one_aged.yaml`, `amount_married_two_aged.yaml`, and `amount_single.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-ae0808f54eaad8ed98d212a5d3ebca73bc80756fefb74de115c8a9a498dfff97R1-R28), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-07e0286e73459a123765c8fb84126c56a35a86f4b8c04578aeb5d8804df9efd0L1-R27), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-bb8128753953cffe82c1a012ad1f7ed841fc234811b52f6dee5ece380c850242R1-R24))
  - Provide description, metadata, and references for each parameter file ([link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-f57a0f95753fda19fbb6ea268cc5ed74763f7735ba320543931bc81c2228328cR1-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-ae0808f54eaad8ed98d212a5d3ebca73bc80756fefb74de115c8a9a498dfff97R1-R28), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-07e0286e73459a123765c8fb84126c56a35a86f4b8c04578aeb5d8804df9efd0L1-R27), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-bb8128753953cffe82c1a012ad1f7ed841fc234811b52f6dee5ece380c850242R1-R24))
  - Implement the state-level tax credit in the policy engine model ([link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-f57a0f95753fda19fbb6ea268cc5ed74763f7735ba320543931bc81c2228328cR1-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-ae0808f54eaad8ed98d212a5d3ebca73bc80756fefb74de115c8a9a498dfff97R1-R28), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-07e0286e73459a123765c8fb84126c56a35a86f4b8c04578aeb5d8804df9efd0L1-R27), [link](https://github.com/PolicyEngine/policyengine-us/pull/2435/files?diff=unified&w=0#diff-bb8128753953cffe82c1a012ad1f7ed841fc234811b52f6dee5ece380c850242R1-R24))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/in/tax/income/credits/unified_tax_credit_for_elderly/age_eligibility.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/in/tax/income/credits/unified_tax_credit_for_elderly/age_eligibility.yaml:1]: need legal code

File: policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_income.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    label = "Indiana unified elderly tax credit income"
+    unit = USD
+    definition_period = YEAR
+    documentation = (
+        "Income that is utilized for the elderly credit calculation"
+    )
+    reference = "https://forms.in.gov/Download.aspx?id=15394 "
+    defined_for = StateCode.IN
+
+    def formula(person, period, parameters):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_income.py:16]: make this a list parameter instead and then use `adds="gov.states.in.tax.income.credits.unified_elderly.income_sources"`

don't need a unit test file then

File: policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        total_income = tax_unit.sum(income)
+        age_head = tax_unit("age_head", period)
+        spouse_age = tax_unit("age_spouse", period)
+        filing_status = tax_unit("filing_status", period)
+        status = filing_status.possible_values
+        married = status.JOINT
+        head_eligible = age_head >= p.age_eligibility
+        spouse_eligible = spouse_age >= p.age_eligibility
+        both_eligible = head_eligible and spouse_eligible
+        one_eligible = head_eligible | spouse_eligible
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py:30]: convention, also `one_eligible` suggests only one
```suggestion
        eligible = head_eligible | spouse_eligible
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        income = person("in_unified_elderly_tax_income", period)
+        total_income = tax_unit.sum(income)
+        age_head = tax_unit("age_head", period)
+        spouse_age = tax_unit("age_spouse", period)
+        filing_status = tax_unit("filing_status", period)
+        status = filing_status.possible_values
+        married = status.JOINT
+        head_eligible = age_head >= p.age_eligibility
+        spouse_eligible = spouse_age >= p.age_eligibility
+        both_eligible = head_eligible and spouse_eligible
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py:29]:
```suggestion
        both_eligible = head_eligible & spouse_eligible
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        one_eligible = head_eligible | spouse_eligible
+        married_amount = one_eligible * where(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py:31]:
```suggestion
        married_amount = where(
```

File: policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            both_eligible,
+            p.amount.married.two_aged.calc(total_income),
+            p.amount.married.one_aged.calc(total_income),
+        )
+        single_amount = p.amount.single.calc(total_income) * head_eligible
+        return where(married, married_amount, single_amount)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/in/tax/income/credits/elderly_tax_credit/in_unified_elderly_tax_credit.py:37]:
```suggestion
        return eligible * where(married, married_amount, single_amount)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**